src/webgpu/shader/execution/expression/call/builtin/subgroupAll.spec.ts

```diff
@@ -0,0 +1,401 @@
+  width: number,
+  height: number
+): Error | undefined {
+  const { blockWidth, blockHeight, bytesPerBlock } = kTextureFormatInfo[format];
+  const blocksPerRow = width / blockWidth;
+  // 256 minimum comes from image copy requirements.
+  const bytesPerRow = align(blocksPerRow * (bytesPerBlock ?? 1), 256);
+  const uintsPerRow = bytesPerRow / 4;
+  const uintsPerTexel = (bytesPerBlock ?? 1) / blockWidth / blockHeight / 4;
+
+  const expected = new Map<number, number>();
+  for (let row = 0; row < height; row++) {
+    for (let col = 0; col < width; col++) {
+      const offset = uintsPerRow * row + col * uintsPerTexel;
+      const subgroup_id = data[offset + 1];
+
+      if (subgroup_id === 0) {
+        return new Error(`Internal error: helper invocation at (${col}, ${row})`);
+      }
+
+      let v = expected.get(subgroup_id) ?? 1;
+      // First index of input is an atomic counter.
+      v &= input[1 + row * width + col];
+      expected.set(subgroup_id, v);
+    }
+  }
+
+  for (let row = 0; row < height; row++) {
+    for (let col = 0; col < width; col++) {
+      const offset = uintsPerRow * row + col * uintsPerTexel;
+      const res = data[offset];
+      const subgroup_id = data[offset + 1];
+
+      if (subgroup_id === 0) {
+        // Inactive in the fragment.
+        continue;
+      }
+
+      const expected_v = expected.get(subgroup_id) ?? 0;
+      if (expected_v !== res) {
+        return new Error(`Row ${row}, col ${col}: incorrect results:
+- expected: ${expected_v}
+-      got: ${res}`);
+      }
+    }
+  }
+
+  return undefined;
+}
+
+g.test('fragment')
+  .desc('Tests subgroupAll in fragment shaders')
+  .params(u =>
+    u
+      .combine('size', kFramebufferSizes)
+      .beginSubcases()
+      .combine('case', [...iterRange(kNumCases, x => x)])
+      .combineWithParams([{ format: 'rg32uint' }] as const)
+  )
+  .beforeAllSubcases(t => {
+    t.selectDeviceOrSkipTestCase('subgroups' as GPUFeatureName);
+  })
+  .fn(async t => {
+    const prng = new PRNG(t.params.case);
+    // Case 0 is all 0s.
+    // Case 1 is all 1s.
+    // Other cases are filled with random 0s and 1s.
+    //
+    // Note: the first index is used as an atomic counter for subgroup ids.
+    const numInputs = t.params.size[0] * t.params.size[1] + 1;
+    const inputData = new Uint32Array([
+      ...iterRange(numInputs, x => {
+        if (x === 0) {
+          // All subgroup ids start from index 1.
+          return 1;
+        } else if (t.params.case === 0) {
+          return 0;
+        } else if (t.params.case === 1) {
+          return 1;
+        }
+        return prng.uniformInt(2);
+      }),
+    ]);
+
+    const fsShader = `
+enable subgroups;
+
+struct Inputs {
+  subgroup_id : atomic<u32>,
+  data : array<u32>,
+}
+
+@group(0) @binding(0)
+var<storage, read_write> inputs : Inputs;
+
+@fragment
+fn main(
+  @builtin(position) pos : vec4f,
+) -> @location(0) vec2u {
+  var subgroup_id = 0u;
+  if subgroupElect() {
```

Comment:
I recommend avoiding divergence here.
The subgroup representative IDs don't have to be sequential.  So it's ok to delete the if condition; it's ok for every invocation to bump the counter.  The broadcast on line 372 still sends the representative ID to all the invocations in the subgroup.